apps/rush-lib/src/data/EnvironmentConfiguration.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+enum EnvironmentVariableNames {
+  TempDirectoryOverride = 'RUSH_TEMP_FOLDER'
+}
+
+const NORMALIZED_ENV_VALUE_NAMES: { [uppercaseValue: string]: EnvironmentVariableNames } = {};
+for (const envValue in EnvironmentVariableNames) {
+  if (EnvironmentVariableNames.hasOwnProperty(envValue)) {
+    const enumValue: string = EnvironmentVariableNames[envValue];
+    NORMALIZED_ENV_VALUE_NAMES[enumValue.toUpperCase()] = enumValue as EnvironmentVariableNames;
+  }
+}
+
+/**
+ * Provides Rush-specific environment variable data. All Rush environment variables must start with "RUSH_". This class
+ * is designed to be used by RushConfiguration.
+ *
+ * @remarks
+ * Initialize will throw if any unknown parameters are present.
+ */
+export class EnvironmentConfiguration {
+  private static _hasBeenInitialized: boolean = false;
+
+  private static _rushTempDirOverride: string | undefined;
+
+  /**
+   * An override for the common/temp directory path.
+   */
+  public static get rushTempDirOverride(): string | undefined {
+    EnvironmentConfiguration._ensureInitialized();
+    return EnvironmentConfiguration._rushTempDirOverride;
+  }
+
+  /**
+   * Reads and validates environment variables. If any are invalid, this function will throw.
+   */
+  public static initialize(): void {
+    EnvironmentConfiguration.reset();
+
+    const unknownEnvVariables: string[] = [];
+    for (const envVar in process.env) {
+      if (process.env.hasOwnProperty(envVar) && envVar.match(/^RUSH_/i)) {
+        const resolvedValueName: EnvironmentVariableNames | undefined = NORMALIZED_ENV_VALUE_NAMES[envVar];
```

Comment:
>envVar [](start = 99, length = 6)

I think you mean envVar.toUpper() here?

Reply:
The NodeJS docs say that process.env[] is case-sensitive on Unix, but case-sensitive for Windows.  It would be helpful to add a comment somewhere explaining that this NORMALIZED_ENV_VALUE_NAMES table is trying to preserve those expectations.

---
In reply to: [178238619](https://github.com/Microsoft/web-build-tools/pull/585#discussion_r178238619) [](ancestors = 178238619)